data/cube/Query.ts

```diff
@@ -119,19 +129,24 @@ export class Query {
     //------------------------
     // Implementation
     //------------------------
-    private parseFields(names: string[]): CubeField[] {
+    private parseFields(raw: CubeField[]|string[]): CubeField[] {
         const {fields} = this.cube;
-        return names ? fields.filter(f => names.includes(f.name)) : fields;
+        if (!raw) return fields;
+        if (raw[0] instanceof CubeField) return raw as CubeField[];
```

Comment:
Is it safe to assume that because of the first instance was a CubeField, all the others will be? I know that's how we've typed it, but in a non-TS environment devs might easily end up with mixed array. Same below for dimensions.

Reply:
could alternatively just make the usage of passing a cube field array an undocumented internal feature, and not even include CubeField[] in the documented type....

Reply:
I'm sold :)